### ISSUE TITLE
⏱️ Call to action: Help shape the future for ASP.NET Core

### ISSUE THREAD

**Issue by mkArtakMSFT** [dotnet/aspnetcore#26625]
Hi. Thanks for stopping by.

We're actively working on .NET 6 planning and we would like your help with this.
In the next few weeks we are going to scan through all the issues in our backlog and identify candidates for .NET 6. To make a decision whether an issue is a good candidate or not for the upcoming release, we also look into the number of upvotes 👍 (and other reactions). To help us prioritize the issues the community feels are most important, please find the issues you want to see resolved and upvote 👍 them.

**⚠️ Please don't add descriptions of what you want to see as comments in this issue. Simply find an existing of file a new issue and upvote it**

**Comment by boukenka** [comment]
AoT compilation https://github.com/dotnet/aspnetcore/issues/5466

**Comment by boukenka** [comment]
SVG Support in Blazor https://github.com/dotnet/aspnetcore/issues/18271

**Comment by JulienM28** [comment]
HTML Autofocus supprt in Blazor WASM

**Comment by JulienM28** [comment]
Projet Reload on save with Kestrel like IIS Express does.

**Comment by glararan** [comment]
#26091 Download compressed resources by default

**Comment by 199621616** [comment]
MobileBlazorbindings in the experiment has got a good response. We hope that it can become a formal project of Microsoft as soon as possible, and can provide complete functions, so that users can focus on their own business logic, especially the hybrid applications. It seems to have a chance to surpass electron, more streamlined but more powerful.

**Comment by rogihee** [comment]
Editor & debugger improvements and Hot Reload (Edit & Continue)

**Comment by expcat** [comment]
#11558 
Streaming API support to MVC

**Comment by mkArtakMSFT** [comment]
All this is great, folks. Please make sure to upvote on the related issues, rather than linking issues here. That's what we're going to look at after all.

**Comment by expcat** [comment]
@mkArtakMSFT #11558 is locked, we can't to upvote.

**Comment by papyr** [comment]
Feature Request: **`Drop-in component for ASP Identity Administration, Portals, UI, SSO, Claims, Token, Multi.Tenant Management`** so that the Visual Studio templates don't require commercial licensed 3rd party components.

I have been watching/waiting few yrs. patiently, however some recent changes have forced me to request -- Microsoft revisit this for their developer community, keeping in mind some of the pricing out there compares to the salaries of 10 to 15 developers in Bangladesh, Vietnam, India, Philippines etc.

Core & Basic need, yet very complex and critical.

There's a void in .NET native drop-in solution, unlike the Java, PHP Stacks etc, where there are many native drop in Identity Administration & Management Frameworks options that are leveraged across all their platforms seamlessly by developers, for e.g. the J2EE Pluggable Identity Management Framework or JSR 168 or JSR 286 or JSR-351 or JSR-375.

Why is this important? because using Java or PHP, it has allowed easy, clear, core and basic functionalities in those native stacks. As a result if you look JSR 168 or JSR 286 or JSR-351 or JSR-375 Multi-tenants, Group to roles, or to claims vice versa is so easy vs. NET , mobile devices, portals, they all work seamlessly and cohesively with security fixes managed by either IBM or SalesForce or Oracle or Red Hat etc. This is enables developer to be productive right out of the gate.

In .Net there is void/very limited support, always requiring a combination of 3rd parties as a barrier to entry & adoption of ASP app. This is non-trivial for developers and security vulnerability that requires the attention of Microsoft Experts.

Example: We have private information sharing site non OSS for the community almost free (pay if you want), and when we started with web forms, then Simple Membership, the Identity, Identity 2 ASP MVC we had implement much of it on top of these from scratch, when we moved to .NET Core it was another effort. Besides the development there was a lot of confusion on the internal concepts and how they now meant new things. Roles, Claims, Federation, SAML then SAML 2.0 and then Open ID 2.

Describe the solution you'd like

- A drop-in is extensible solution that supports ASP Identity eco-system, Administration, UI, SSO, Token, Multi-tenant Management
- A configuration section to turn on-off the various features.
- Embedded into VS Templates (remove Identity Server etc.)
- Additional context
- The current options and the pricing eliminate many of the existing applications from continuing usage on the .NET stack without extensive retooling or completely abandoning the MS framework.

Its high time MS address this core gate-keeping feature!!

**Comment by papyr** [comment]
- [ ] Controller/API `DDOS throttling` support
- [ ] Login Controller/API Captcha Anti Spambot support
- [ ] Timeouts are easily by passed using Singal-R and there's [no way to force anonymous connections to disconnect](https://stackoverflow.com/questions/49590384/signalr-persist-connections-for-anonymous-users)
- [ ] No way to handle anonymous chats on public website. For e.g. if _Web User is Surfing/browsing for help or FAQ on product/chat_ on ASP website with SignalR, we have to create a duct-taped solution for this, there is no way to identify them uniquely to create a chat stream/session unless they login.

**Comment by mkArtakMSFT** [comment]
> @mkArtakMSFT #11558 is locked, we can't to upvote.

@expcat I've reopened it!

**Comment by mkArtakMSFT** [comment]
@papyr please upvote to the related issues and if you can't find such, create new issues for each one separately. That way we can actually track these requests. Otherwise many asks will be lost / forgotten.

**Comment by gulshan** [comment]
Built-in admin panel template and Vue integration.

**Comment by papyr** [comment]
@mrkarMSFT thanks for the idea 👍 but they keep closing my feature requests and locking them, so its not able to upvote!

**Comment by papyr** [comment]
> Built-in admin panel template and Vue integration.

Do you mean Like Skoruba.Admin then upvote https://github.com/dotnet/aspnetcore/issues/26594

**Comment by mnmr** [comment]
RFC 7692 (WebSocket per-message compression) for SignalR. It seems this depends on [runtime#20004](https://github.com/dotnet/runtime/issues/20004) but it seems relevant in this context. There was an old [issue](https://github.com/aspnet/WebSockets/issues/19) referencing this, but it is now closed and I could find no equivalent open issue here, so I am recommending that people upvote on the runtime issue.

**Comment by 3GDXC** [comment]
blazor wasm-to-wasm interop without javascript (WASI)  [(https://github.com/WebAssembly/interface-types/blob/master/proposals/interface-types/Explainer.md)]

**Comment by 3GDXC** [comment]
please provide a way to improve the user experience for blazor wasm load progress, progress bar and/or filename currently downloading.

**Comment by 3GDXC** [comment]
project template blazor server/wasm using fast (components) as a user interface as quick start/example

**Comment by 3GDXC** [comment]
please add support for sftp

**Comment by seangwright** [comment]
"Proposal: View Component Slots" is locked

> Provide a built-in or otherwise framework supplied mechanism for View Components, when rendered via a tag helper, to render user-supplied child content that can be injected into the View Component's view template in pre-defined locations

https://github.com/dotnet/aspnetcore/issues/4901

**Comment by saint4eva** [comment]
In-house (Microsoft made) identity server and other security modules

**Comment by valeriob** [comment]
I can't find an issue regarding fast dev loop in general that is not mentioning blazer, I would like the whole dev experience  would be much faster

**Comment by mrpmorris** [comment]
Might it be a good idea to create a new call to action with comments disabled?
@mkArtakMSFT

**Comment by mnmr** [comment]
@mrpmorris @mkArtakMSFT You should seriously consider abandoning the whole call-to-action thing. Choosing what issues need solving based on community votes is why I never leave MS product feedback/bug reports anymore - if you don't have a social following or plenty of co-workers to upvote your issues they just languish and eventually get closed.

My suggestion would be to select topics / general directions that the project can move in, let people pick from those, and let that guide part of the overall effort, without making specific commitments to specific issues.

**Comment by mrpmorris** [comment]
It's a non-binding advisory vote :)

**Comment by mkArtakMSFT** [comment]
> @mrkarMSFT thanks for the idea 👍 but they keep closing my feature requests and locking them, so its not able to upvote!
> 
> Can you please [unlock 26594](https://github.com/dotnet/aspnetcore/issues/26594) to allow voting on the feature please!

@papyr looking into the issue I believe it's not something aligned with our long-term vision. And that's the reason why @blowdart has closed that issue. Given that, it seems we won't be able to prioritize that one for .NET 6 for sure.

**Comment by mkArtakMSFT** [comment]
@3GDXC you've listed multiple feature requests above. Please make sure you file separate issues for each one so we can track these better.

**Comment by mkArtakMSFT** [comment]
> Might it be a good idea to create a new call to action with comments disabled?
> @mkArtakMSFT

I was considering that originally, but didn't want to mute people. If this continues, I will probably lock this conversation.

**Comment by mkArtakMSFT** [comment]
> @mrpmorris @mkArtakMSFT You should seriously consider abandoning the whole call-to-action thing. Choosing what issues need solving based on community votes is why I never leave MS product feedback/bug reports anymore - if you don't have a social following or plenty of co-workers to upvote your issues they just languish and eventually get closed.
> 
> My suggestion would be to select topics / general directions that the project can move in, let people pick from those, and let that guide part of the overall effort, without making specific commitments to specific issues.

We're trying to get the community be vocal about what they want. This and what you suggest are not conflicting efforts, rather extending each other. Eventually, after we are done with the first phase of our planning, we will put together a high-level list of things we will consider for .NET 6. This is what we've done for Blazor during last few major releases (including the 5.0 which is finalizing now).

**Comment by RChrisCoble** [comment]
> AoT compilation #5466

AOT combined with browser task based parallelism would destroy JS.

**.NET WebAssembly support of task based parallelism. #40619**
https://github.com/dotnet/runtime/issues/40619

**Comment by anthcool** [comment]
Make it stupid simple to debug Blazor Server apps that throw circuit errors stating only to turn on detailed exceptions, but when done, don't give any better information after the fact.

**Comment by mkArtakMSFT** [comment]
> Make it stupid simple to debug Blazor Server apps that throw circuit errors stating only to turn on detailed exceptions, but when done, don't give any better information after the fact.

@anthcool  please file an issue for this. We will triage it and include in our planning.

**Comment by anthcool** [comment]
> > Make it stupid simple to debug Blazor Server apps that throw circuit errors stating only to turn on detailed exceptions, but when done, don't give any better information after the fact.
> 
> @anthcool please file an issue for this. We will triage it and include in our planning.

Just added one.  Thanks!

**Comment by ylr-research** [comment]
I think it'd be really useful to have built-in support for Pub/Sub, or at least something close to .NET events. Today scenarios like, having component B (child) and component A (parent), notify components C...N (not child of A), are not that easy to do. Technically, you need to come up with a hack.

**Comment by ylr-research** [comment]
This issue was closed and still there is no solution #26450. It has been also mentioned in [here](azure-cosmos-dotnet-v3/issues/1865) and [here](https://github.com/dotnet/runtime/issues/40074). Since .NET 5 RC1 was released, Blazor WASM stopped working with Azure.

**Comment by mkArtakMSFT** [comment]
> I think it'd be really useful to have built-in support for Pub/Sub, or at least something close to .NET events. Today scenarios like, having component B (child) and component A (parent), notify components C...N (not child of A), are not that easy to do. Technically, you need to come up with a hack.

@ylr-research please file a separate issue to continue the conversation there.

**Comment by ElderJames** [comment]
Automatic module separation and lazy loading.

**Comment by VenkateshSrini** [comment]
Render conditional closing and opening tag https://github.com/dotnet/aspnetcore/issues/26368. With out most of the excellent layouts are not possible to be rendered in Blazor

**Comment by VenkateshSrini** [comment]
Add all the properties of HttpMessageHandler on the client side. At present, it has lots of missing code and we are unable to use many of them

**Comment by pdevito3** [comment]
#26487: New WebHostFactory option to directly call a specific testing startup class when working with integration tests

**Comment by fnandocontreras** [comment]
Use of InMemoryCache in Blazor wasm

**Comment by lloydjatkinson** [comment]
Support for feature folders either out of the box or in a nuget. Instead of the typical messy:

Arranging by types:
```
Controllers
    CustomerController
    InvoiceController
    SmsController
    ...hundreds of files
Views
    CustomerUpdate.cshtml
    CustomerDelete.cshtml
    InvoiceView.cshtml
    SmsSend.cshtml
    ...hundreds of files
```

Arranging by features instead:
```
Features
    Sms
        ...relevant files here
    Customers
        ...relevant files here
    Invoice
        ...relevant files here
```

Read more about it here: https://dev.to/jamesmh/the-life-changing-and-time-saving-magic-of-feature-focused-code-organization-1708

This works for MVC, Razor Pages, Rest APIs.

**Comment by VaclavElias** [comment]
Nested Partial and Simple Partial Child Content 

https://github.com/dotnet/aspnetcore/issues/24470

**Comment by mrpmorris** [comment]
Support older Edge with polyfills
https://github.com/dotnet/aspnetcore/issues/26475

**Comment by OmidAtaei1362** [comment]
Hi. Cascade state for parent child components.

**Comment by ylr-research** [comment]
> > I think it'd be really useful to have built-in support for Pub/Sub, or at least something close to .NET events. Today scenarios like, having component B (child) and component A (parent), notify components C...N (not child of A), are not that easy to do. Technically, you need to come up with a hack.
> 
> @ylr-research please file a separate issue to continue the conversation there.

See #26728

**Comment by craigajohnson** [comment]
SASS as first-class citizen in the toolchain (recognition of variables, intellisense across all imports)

**Comment by craigajohnson** [comment]
Direct support for hosting a Blazor app on a desktop, with access to local peripherals and .NET Core on the client. WASM is great but if we have the full desktop available it makes things like point-of-sale functionality to local devices that much easier

**Comment by som-nitjsr** [comment]
not sure when you will fix this for .Net core 
https://github.com/dotnet/runtime/issues/29935

**Comment by rogihee** [comment]
Drag & drop.  Would be nice to finish McKinnon's work: [https://github.com/dotnet/aspnetcore/issues/18754](https://github.com/dotnet/aspnetcore/issues/18754)

**Comment by shaggygi** [comment]
Bedrock Futures: #15005

**Comment by OmidAtaei1362** [comment]
Cascade Components - Detect parent component automatically

**Comment by YohanSciubukgian** [comment]
Official support for HTTP/3 (QUIC) & in sync with Azure services ideally ;)

**Comment by LukeTOBrien** [comment]
One thing that's been bugging me lately is reducing download size see #26780  
It's just a little thing, but removing dependencies makes all the difference 😉

**Comment by jotosmurf** [comment]
Improve prerendering #26794 (Avoid absolute classic FOOC effect.)

**Comment by yehia2amer** [comment]
Official Microsoft Design components for Blazor: #11229

**Comment by mrlife** [comment]
Built-in way to get a child or parent component reference in Blazor: https://github.com/dotnet/aspnetcore/issues/26826

**Comment by RichardD2** [comment]
SSRS support: #362, #1528, #1764, #12666, #22304

All issues have been closed and locked, so cannot be upvoted.

Currently 1005 votes on the feedback site, with no update since 2018:
[develop a SSRS ReportViewer for ASP.NET Core – Customer Feedback for ACE Community Tooling](https://feedback.azure.com/forums/908035-sql-server/suggestions/33241936-develop-a-ssrs-reportviewer-for-asp-net-core)

**Comment by rodmac1** [comment]
I believe there’s a big gap 3rd party control vendors are struggling to fill e.g. there’s a need for a responsive menu control which is vertical on mobile and horizontal on desktop. Equally it needs to be multi-level and or templated. I’ve found it a struggle altering Blazor’s menu and looking at StackOverflow, I’m not alone. Currently the 3rd party control vendors are fixed on things like grids for the finance industry. This one’s pretty essential to get devs started on creating intuitive and responsive apps that work both for mobile and desktop.

**Comment by rodmac1** [comment]
I would like to see a best in class UI design tool aka 'MAUI Studio' for building HTML/SVG/XAML front ends, something akin to Flutter and supported by graphics design software companies e.g. Corel Corporation and Adobe.

**Comment by mrpmorris** [comment]
I'd like more app-level control over rendering https://github.com/dotnet/aspnetcore/issues/26016

**Comment by mrpmorris** [comment]
@danroth27 what happened with the issue of Blazor being blocked by some types of router? Other than providing a manual workaround is anything being done to work around this? The tickets won't allow new comments.

https://github.com/dotnet/aspnetcore/issues/19511

https://github.com/dotnet/aspnetcore/issues/19552

**Comment by ElderJames** [comment]
Static site generator #26929

**Comment by danroth27** [comment]
@mrpmorris If you're in an environment where files with "dll" extensions are simply blocked, then we've provided manual steps for changing the extension. We haven't identified any additional work in this area currently. If you think we need to do more, please file an issue with your suggestion and we'll take a look.

**Comment by PercyLeon123** [comment]
Good morning, I hope I am not mistaken but I have noticed that the Timer () is
slows down when you switch tabs in the browser.
Make a basic example of the possible problems and this is my repository link [(https://github.com/PercyLeon123/Cronometro).](https://github.com/PercyLeon123/Cronometro)
To detail what happened, carry out some tests and record it.

Basic and simple example, slows down 10 minutes [https://youtu.be/jHLVCSGXPMs](https://youtu.be/jHLVCSGXPMs)
Example with more code, it slows down 1 minute and 50 seconds [https://youtu.be/YhEVTVB2ai0](https://youtu.be/YhEVTVB2ai0)
if you do not change the tab the timer does not slow down [https://youtu.be/2NI0NhRLdFY](https://youtu.be/2NI0NhRLdFY)

In the description I detail what happened. I'm Percy León from Lima-Peru
Contact me [Linkedin](https://www.linkedin.com/in/percy-luis-leon-lucas/)
Greetings!

**Comment by mkArtakMSFT** [comment]
Thanks for reporting this, @PercyLeon123.
Can you please file a new issue to track this separately.

**Comment by RehanSaeed** [comment]
Most of the feature requests seem to be Blazor related. While this is understandable due to it being a new technology, with respect to my fellow Blazor devs there are still those of us who don't use Blazor for various reasons that are not important. I hope that development time is not biased too much in favour of that technology to the detriment of core ASP.NET features that would help everyone due to their being core features. Here are the top features according to upvotes:

-  Add support for LetsEncrypt  ([LettuceEncrypt](https://github.com/natemcmaster/LettuceEncrypt) still needs DNS renewal) #4712
-  Add HTTP/3 Support #15271 
-  Add Streaming API Support to MVC #11558 
- Add HTTP/2: Server Push Support #4249 
- Define a JSON schema for appsettings.json #2867
- `Async` suffix for controller action names will be trimmed by default #8998 
- `ProblemDetails` is not returned for 404 NotFound and 500 #4953 
- Update ASP.NET Core to use C# 8's nullable reference types #5680 
- Add IAsyncStartup and IAsyncStartupFilter to support async startup classes and filters #5897 
- ConfigureTestContainer not working with GenericHost #14907 
- Bedrock Endgame #15005
- Add support for content security policy #6001 
- Support the new `Forwarded` header (RFC 7239) #5978 
- Make it easy to configure a host/scheme for absolute URLs with LinkGenerator #14192
- New WebHostFactory option to directly call a specific testing startup class when working with integration tests #26487
-  JsonPatchDocument should use System.Text.Json in ASP.NET vNext #24333 

You can see a full list of non-Blazor issues [here](https://github.com/dotnet/aspnetcore/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc+-label%3Aarea-blazor).

**Comment by tbonham** [comment]
> Feature Request: **`Drop-in component for ASP Identity Administration, Portals, UI, SSO, Claims, AD-Sync, Token, Multi.Tenant Management`** so that the Visual Studio templates don't require commercial licensed 3rd party components.
> 
> I have been watching/waiting few yrs. patiently, however some recent changes have forced me to request -- Microsoft revisit this for their developer community, keeping in mind some of the pricing out there compares to the salaries of 10 to 15 developers in Bangladesh, Vietnam, India, Philippines etc.
> 
> Core & Basic need, yet very complex and critical.
> 
> There's a void in .NET native drop-in solution, unlike the Java, PHP Stacks etc, where there are many native drop in Identity Administration & Management Frameworks options that are leveraged across all their platforms seamlessly by developers, for e.g. the J2EE Pluggable Identity Management Framework or JSR 168 or JSR 286 or JSR-351 or JSR-375.
> 
> Why is this important? because using Java or PHP, it has allowed easy, clear, core and basic functionalities in those native stacks. As a result if you look `JSR 168` or `JSR 286` or `JSR-351` or `JSR-375` Multi-tenants, [Group to roles, or to claims](https://javaee.github.io/security-spec/spec/jsr375-spec.html#_group_to_role_mapping) vice versa is so easy vs. NET , mobile devices, portals, they all work seamlessly and cohesively with security fixes managed by either IBM or SalesForce or Oracle or Red Hat etc. This is enables developer to be productive right out of the gate.
> 
> In .Net there is void/very limited support, always requiring a combination of 3rd parties as a barrier to entry & adoption of ASP app. This is non-trivial for developers and security vulnerability that requires the attention of Microsoft Experts.
> 
> Example: We have private information sharing site non OSS for the community almost free (pay if you want), and when we started with web forms, then Simple Membership, the Identity, Identity 2 ASP MVC we had implement much of it on top of these from scratch, when we moved to .NET Core it was another effort. Besides the development there was a lot of confusion on the internal concepts and how they now meant new things. Roles, Claims, Federation, SAML then SAML 2.0 and then Open ID 2.
> 
> Describe the solution you'd like
> 
> * A drop-in is extensible solution that supports ASP Identity eco-system, Administration, UI, SSO, Token, Multi-tenant Management
> * A configuration section to turn on-off the various features.
> * Embedded into VS Templates (remove Identity Server, use native MS option etc.)
> * Allow-Easy AD Active Directory Integration
> * `User Defined/Created Fields UDF` from the Admin level Support.
> * The current options and the pricing eliminate many of the existing applications from continuing usage on the .NET stack without extensive retooling or completely abandoning the MS framework.
> 
> Its high time MS address this core gate-keeping feature!!

We need SAML2 and CAS add to.

**Comment by papyr** [comment]
Auto Upgrade option **PartialViews to [ViewComponents](https://github.com/dotnet/aspnetcore/issues/27007#issue-723796935)**  Lots of legacy code is hung in the partials, we need to port that to the new `ViewComponents` or `TagHelpers `apps

**Comment by YohanSciubukgian** [comment]
Concurrent frameworks can support a wider range of architectures (not sure if this list is exhaustive):
 - NodeJs support : x86 / x64 / arm32 / arm64 / ppc64le / ppc64be / x390x
 - OpenJDK support : x86 / x64 / arm32 / arm64 / ppc32 / ppc64le / s390x / sparcv9 
 - AspNet core support : x89 / x64 / arm32 / arm64

please add support to more architectures such as ppc (32 & 64) : https://github.com/dotnet/runtime/issues/10055

**Comment by richlander** [comment]
Thanks for the feedback @YohanSciubukgian. Our plan is to support most/all mainline OSes and chip combinations. We now have that. At present, the only one we're really missing is Linux x86 (32-bit). That comes up quite seldom. The additional ones that you've mentioned come up even less. The community is free to add the more niche configurations that you'd mention, but I'm doubtful that will happen. They don't, AFAICT, have any practical commercial value. 

Personally, I'd much rather we added support for FreeBSD (on x64) than PPC. I suspect that would benefit an order of magnitude more users (if not much more).

Also note that we're currently putting a fair bit of effort into supporting Apple Silicon and Arm64 (generally). Just supporting new chips takes a lot of investment. Context: https://github.com/dotnet/runtime/issues/43313 and https://devblogs.microsoft.com/dotnet/arm64-performance-in-net-5/. It is much better to focus on doing a great job for mainline chips and leave the legacy/hobbyist chips for other platforms to support.

There is a company working on MIPS support for .NET Core currently: https://github.com/dotnet/runtime/issues/38069.

This HN post is super related: https://news.ycombinator.com/item?id=24640545.

**Comment by YohanSciubukgian** [comment]
> Thanks for the feedback @YohanSciubukgian. Our plan is to support most/all mainline OSes and chip combinations. We now have that. At present, the only one we're really missing is Linux x86 (32-bit). That comes up quite seldom. The additional ones that you've mentioned come up even less. The community is free to add the more niche configurations that you'd mention, but I'm doubtful that will happen. They don't, AFAICT, have any practical commercial value.
> 
> Personally, I'd much rather we added support for FreeBSD (on x64) than PPC. I suspect that would benefit an order of magnitude more users (if not much more).
> 
> Also note that we're currently putting a fair bit of effort into supporting Apple Silicon and Arm64 (generally). Just supporting new chips takes a lot of investment. Context: [dotnet/runtime#43313](https://github.com/dotnet/runtime/issues/43313) and https://devblogs.microsoft.com/dotnet/arm64-performance-in-net-5/. It is much better to focus on doing a great job for mainline chips and leave the legacy/hobbyist chips for other platforms to support.
> 
> There is a company working on MIPS support for .NET Core currently: [dotnet/runtime#38069](https://github.com/dotnet/runtime/issues/38069).
> 
> This HN post is super related: https://news.ycombinator.com/item?id=24640545.

Thanks for your answer. I don't know if it's possible but as you had merged mono in dotnet/runtime repo for .net 5, I had in mind to expose existing architectures initially developped for mono but with official Microsoft support.

**Comment by richlander** [comment]
Great question. The Mono in dotnet/runtime is targeted at Android, iOS, and wasm.

**Comment by JulienM28** [comment]
Built in Identity Server #27053

**Comment by kursatarslan** [comment]
IdentityServer Token Manager

**Comment by mahdisml** [comment]
WebSocket performance improvement

**Comment by ivanivanyuk1993** [comment]
Multithreading
https://github.com/dotnet/runtime/issues/40619

**Comment by daniel-p-tech** [comment]
JS isolation improvement - Colocate CS, CSS and JS files

https://github.com/dotnet/aspnetcore/issues/26145

**Comment by davidfowl** [comment]
@mahdisml please file an issue with relevant details.

**Comment by saifxx** [comment]
Enterprise UI Components for Blazor

**Comment by mrlife** [comment]
Publish Blazor Server or WASM to iOS/Android #27252

**Comment by mkArtakMSFT** [comment]
> I can't find an issue regarding fast dev loop in general that is not mentioning blazor, I would like the whole dev experience would be much faster

Here you go: https://github.com/dotnet/aspnetcore/issues/18486

**Comment by VenkateshSrini** [comment]
Compensating transactions are a very tricky part of micro services development. Can we come up with some sort of high level framework that will facilitate it. The lower level implementation say for example an implementation with redis or RabbitMQ can then be provided my community to augment it.

**Comment by mrlife** [comment]
Blazor InputSelect support for `multiple` attribute. https://github.com/dotnet/aspnetcore/issues/5519

**Comment by mikeandersun** [comment]
Decuple the `SignInManager `and `UserManager ` in Identity.
[#22597](https://github.com/dotnet/aspnetcore/issues/22597)

**Comment by osintsevvladimir** [comment]
Dynamically creating and adding components.
Angular `ComponentFactoryResolver` alternative.

```
@Component({
 selector: 'app-root',
 template: `
 <template #parent></template>
 <button (click)="createChild()">Create Child</button> 
 `,
})
export class ParentComponent {
 @ViewChild('parent', { read: ViewContainerRef }) container;

 constructor(private resolver: ComponentFactoryResolver) {}

 createChild() {
 this.container.clear();
 const factory: ComponentFactory = this.resolver.resolveComponentFactory(ChildComponent);
 this.componentRef: ComponentRef = this.container.createComponent(factory);
 }
}
```
